ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -136,23 +134,80 @@ private boolean populateOutput(AtomicLong totalKeysProcessed, Map<String, BlobSt
             == replicaCount)) {
           if (includeAcceptableInconsistentBlobs) {
             logger.info("Partially deleted (acceptable inconsistency) blob " + blobId + " isDeletedOrExpired "
-                + consistencyBlobResult.getIsDeletedOrExpired() + "\n" + consistencyBlobResult);
+                + consistencyBlobResult.isDeletedOrExpired() + "\n" + consistencyBlobResult);
           }
           acceptableInconsistentBlobs++;
         } else {
-          realInconsistentBlobs++;
-          logger.info(
-              "Inconsistent Blob : " + blobId + " isDeletedOrExpired " + consistencyBlobResult.getIsDeletedOrExpired()
-                  + "\n" + consistencyBlobResult);
+          earliestRealInconsistentBlobTimeMs =
+              Math.min(earliestRealInconsistentBlobTimeMs, consistencyBlobResult.getOpTime());
+          if (!(consistencyBlobResult.getAvailable().size() + consistencyBlobResult.getUnavailableList().size()
```

Comment:
i'm not sure I follow this check and why it needs to be so complicated. Why isn't this just `if(!consistencyBlobResult. belongsToRecentIndexSegment())`?